### PR TITLE
feat(messages): now displays the list of current discussions with the…

### PR DESCRIPTION
**Pull request by Sobell13**
Resolving this issue through this Pull Request introduces several new features : 

1. Using the UIDs in the ChatSummary
    The ChatSummary now returns a concatenated string of user UIDs instead of names, enhancing flexibility and supporting multiple users.
    A helper function converts UIDs into display names when needed.
2. Sending UIDs in the Firebase Database
     The repository now stores UIDs in the Firebase database instead of names, ensuring that data is more consistent, more private and reliable.
     This change allows the app to retrieve users by their UIDs when fetching the latest ChatSummary.
3. Displaying the ChatSummary
     The Message Screen now displays a complete preview list of chats, showing relevant information.
     Clicking on any chat preview opens the corresponding Chat Screen.

### CODE REVIEW

File: app/src/test/java/com/swent/suddenbump/model/chat/ChatSummaryTest.kt
Comment (ifaresse) [inline app/src/test/java/com/swent/suddenbump/model/chat/ChatSummaryTest.kt:1]: Why the test dateWorksWithNonNullDate() is commented ? Still useful or can be deleted ?

**Review by ifaresse — CHANGES_REQUESTED** [review]
Look at my comment for the commented test and otherwise the app crashed when I navigate to the list of current discussions.
Capture d'écran 2024-11-23 003530
It displays this error in the log.

File: app/src/test/java/com/swent/suddenbump/model/chat/ChatSummaryTest.kt
Comment (Sobell13) [reply app/src/test/java/com/swent/suddenbump/model/chat/ChatSummaryTest.kt:1]: The test is not useful anymore

**Comment by Sobell13** [comment]
The messages with the previous version of the app did not have the same format as the newest one, making it crash when retrieving old messages.
In order to change that, deleting messages and sending only with the latest version is mandatory.

File: app/src/test/java/com/swent/suddenbump/model/chat/ChatSummaryTest.kt
Comment (ifaresse) [reply app/src/test/java/com/swent/suddenbump/model/chat/ChatSummaryTest.kt:1]: Can you remove it then ?

File: output_diff.txt
Comment (ifaresse) [inline output_diff.txt:1]: what is this file ?

**Review by ifaresse — APPROVED** [review]
LGTM !

File: app/src/test/java/com/swent/suddenbump/model/chat/ChatSummaryTest.kt
Comment (Sobell13) [reply app/src/test/java/com/swent/suddenbump/model/chat/ChatSummaryTest.kt:1]: It's deleted now with the latest commit

File: output_diff.txt
Comment (Sobell13) [reply output_diff.txt:1]: I forgot to delete it.
Should be good now.

**Pull request merged**